src/ruleLoader.ts

```diff
@@ -43,23 +38,19 @@ export function loadRules(ruleOptionsList: IOptions[],
 
     for (const ruleOptions of ruleOptionsList) {
         const ruleName = ruleOptions.ruleName;
-        const enableDisableRules = enableDisableRuleMap.get(ruleName);
-        if (ruleOptions.ruleSeverity !== "off" || enableDisableRuleMap) {
```

Comment:
If you use tslint as a library, you can pass different rule configs for every call to `Linter.lint`.
That's why this needs to be done for every file.
On the other hand, I don't know if that makes sense to keep that "feature".